### PR TITLE
feat: support AI PKs for dm_copy_to() for empty tables (ptype)

### PR DESCRIPTION
**Pull request by TSchiefer**
on SQLite, SQL Server, Postgres

code mostly taken from https://github.com/cynkra/dm/pull/1654

### CODE REVIEW

**Comment by TSchiefer** [comment]
need to think about this trimws

**Comment by TSchiefer** [comment]
pressed the "request review" button a bit early... Now it seems to be ready for review though.

File: R/build_copy_queries.R [hunk 1]
```diff
@@ -22,20 +22,76 @@ build_copy_queries <- function(dest, dm, set_key_constraints = TRUE, temporary =
...
+    df_col_types <-
+      enframe(types, "col", "type") %>%
+      mutate(autoincrement_attribute = "")
+
+    if (length(pk_col) > 0L) {
+      df_col_types <-
+        df_col_types %>%
+        mutate(autoincrement_attribute = autoincrement_attribute)
+    }
+
```
Comment (krlmlr) [inline R/build_copy_queries.R:85]: Is this still necessary?

```suggestion
```

File: R/build_copy_queries.R [hunk 2]
```diff
@@ -22,20 +22,76 @@ build_copy_queries <- function(dest, dm, set_key_constraints = TRUE, temporary =
...
+      # Ref: https://duckdb.org/docs/sql/statements/create_sequence
+
+      # SQLite:
+      # For a primary key, autoincrementing works by default, and it is almost never
+      # necessary to use the `AUTOINCREMENT` keyword. So nothing we need to do here.
+      # Ref: https://www.sqlite.org/autoinc.html
+    }
+    df_col_types <-
+      enframe(types, "col", "type") %>%
+      mutate(autoincrement_attribute = "")
```
Comment (krlmlr) [inline R/build_copy_queries.R:78]: Perhaps:

```suggestion
      mutate(autoincrement_attribute = !!autoincrement_attribute)
```

**Review by krlmlr — APPROVED** [review]
Looks good, please merge and move on after deciding the two mini questions.

**Comment by TSchiefer** [comment]
the autoincrement_attribute is only used for MariaDB and in case pk_col is an AI PK col. I couldn't think of a nicer way to ensure that it's only set under these conditions than [1be9a55](https://github.com/cynkra/dm/pull/1696/commits/1be9a557d7c1c5c16f9eaa48d46296a9b5ae3d43)

**Pull request merged**